### PR TITLE
fix(composition): Remove composition responsibilities from supergraph printer

### PR DESCRIPTION
**Pull request by trevor-scheer**
Push schema construction responsibilities up and out of the supergraph printer and into composition where they belong.

* Update tests (sort related)
* Fix incorrect test for printFederatedSchema (exposed by an unexpected test update)

I encountered these issues during implementation for #996 and figured this work was a separate issue and worth breaking out. #996 will be based against this branch.

### CODE REVIEW

File: federation-js/src/service/printSubgraphSchema.ts [hunk 1]
```diff
@@ -54,7 +58,7 @@ type Options = {
  *
  *    - commentDescriptions:
  *        Provide true to use preceding comments as the description.
- *
+ * @deprecated Use `printSubgraphSchema` instead.
  */
 export function printSchema(schema: GraphQLSchema, options?: Options): string {
```
Comment (martijnwalraven) [inline federation-js/src/service/printSubgraphSchema.ts:63]: Maybe turn this around: make `printSubgraphSchema` the main function, and alias it as `printSchema` for backwards compatibility?

**Review by martijnwalraven — APPROVED** [review]
Good catch! Moving responsibilities out of printing nicely cleans things up.

File: federation-js/src/service/printSubgraphSchema.ts (hunk 1 above)
Comment (trevor-scheer) [reply federation-js/src/service/printSubgraphSchema.ts:63]: Ah, yes of course 😄

Comment (trevor-scheer) [reply federation-js/src/service/printSubgraphSchema.ts]: 5227ad5

**Pull request merged**